### PR TITLE
Refactored energy detection procedure

### PR DESCRIPTION
**Pull request by hubertmis**

### CODE REVIEW

File: src/nrf_drv_radio802154_fsm.c [hunk 1]
```diff
@@ -2837,10 +2941,23 @@ bool nrf_drv_radio802154_fsm_transmit(nrf_drv_radio802154_term_t term_lvl,
...
 
-    bool result = false;
+    if (result)
+    {
+        state_set(RADIO_STATE_ED);
+        m_ed_time_left = time_us;
+        m_ed_result    = 0;
+    }
+
+    if (result)
```
Comment (rlubos) [inline src/nrf_drv_radio802154_fsm.c]: `result` does not seem to be modified inbetween, why separate `if (result)`?

File: src/nrf_drv_radio802154_fsm.c [hunk 2]
```diff
@@ -2837,10 +2941,23 @@ bool nrf_drv_radio802154_fsm_transmit(nrf_drv_radio802154_term_t term_lvl,
...
+    {
+        ed_begin(true);
+    }
+
+    return result;
+
+    result = false;
```
Comment (rlubos) [inline src/nrf_drv_radio802154_fsm.c]: It doesn't look like reachable code. Shouldn't it be removed?

Comment (hubertmis) [reply src/nrf_drv_radio802154_fsm.c]: Currently there is a lot of unreachable code. I'm going to remove it at the end or refactoring. Right now I want to keep it as reference.

Comment (rlubos) [reply src/nrf_drv_radio802154_fsm.c]: I understand that, but how can reviewer tell if something is left by author on purpose or if it is a bug? I would wrap it inside `#if 0 ... #endif`, so there will be no misunderstandigs (like you did before by the way).

File: src/nrf_drv_radio802154_fsm.c (hunk 1 above)
Comment (hubertmis) [reply src/nrf_drv_radio802154_fsm.c]: It will be probably modified by RAAL functions.

**Pull request merged**